crates/forge-runner/src/collection_artifacts.rs

```diff
@@ -0,0 +1,90 @@
+use cairo_felt::Felt252;
+use cairo_lang_sierra::program::Program;
+use cairo_lang_test_plugin::test_config::{PanicExpectation, TestExpectation};
+use serde::Deserialize;
+use starknet::core::types::BlockId;
+use url::Url;
+
+#[derive(Debug, Clone)]
+pub struct CompiledTestCrateRunnable {
+    sierra_program: Program,
+    test_cases: Vec<TestCaseRunnable>,
+}
+
+impl CompiledTestCrateRunnable {
+    #[must_use]
+    pub fn new(sierra_program: Program, test_cases: Vec<TestCaseRunnable>) -> Self {
+        Self {
+            sierra_program,
+            test_cases,
+        }
+    }
+
+    pub(crate) fn test_cases(&self) -> &Vec<TestCaseRunnable> {
+        &self.test_cases
+    }
+
+    pub(crate) fn sierra_program(&self) -> &Program {
+        &self.sierra_program
+    }
+}
```

Comment:
Can I remove this "encapsulation"? 🙏 
cc @cptartur

Reply:
shots fired